src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -397,17 +399,24 @@ public inherited sharing class RD2_OpportunityEvaluationService {
             return false;
         }
 
+        Boolean isCandidate = false;
+
         if (isBatchContext) {
-            return !rdSettings.isDisableAllInstallments;
+            isCandidate = !rdSettings.isDisableAllInstallments;
 
         } else if (rd.isNew()) {
-            return rdSettings.isFirstInstallmentEnabled;
+            isCandidate = rdSettings.isFirstInstallmentEnabled;
 
         } else if (!rd.hasCurrentOpenOpportunity(currentDate)) {
-            return rdSettings.isNextInstallmentEnabled;
+            isCandidate = rdSettings.isNextInstallmentEnabled;
+        }
+
+        if (rd.isFixedLength()) {
```

Comment:
Determining the additional criteria for a fixed-length RD at this point is not required since the Opp service will check if the next installment is required, so this method can stay generic.